mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -556,6 +579,28 @@ class SparseVector(
       i += 1
     }
   }
+
+  override def hashCode(): Int = {
```

Comment:
For dense vectors, it may contains many zeros at the beginning, and then the cost is still high if we look for the first 16 nonzeros. The `hashCode` is not used in any MLlib code. This is just to reduce the overhead of Pyrolite serializer.